embedded/README.md

```diff
@@ -0,0 +1,252 @@
+│
+├── 📁 receiver/
+│   ├── CMakeLists.txt          # Receiver build configuration
+│   ├── 📁 config/              # Receiver-specific config files
+│   ├── 📁 src/                 # Receiver source code
+│   ├── sst.config              # Receiver runtime configuration
+│   └── update-credentials.sh   # Script to update stored credentials
+│
+├── 📁 sender/
+│   ├── CMakeLists.txt          # Sender build configuration
+│   └── 📁 src/                 # Sender source code
+│
+└── 📁 src/
+    ├── cmd_handler.c           # Command processing implementation
+    ├── config_handler.c        # Configuration management logic
+    ├── mbedtls_time_alt.c      # Custom mbedTLS time source
+    ├── pico_handler.c          # Pico helper logic
+    ├── pico_helper.c           # Additional Pico utility functions
+    └── sst_crypto_embedded.c   # Embedded crypto API implementation
+```
+
+## Hardware Requirements
+
+### **Sender (Pico)**
+- [Raspberry Pi Pico (RP2040)](https://www.sparkfun.com/raspberry-pi-pico.html?src=raspberrypi)
+- Li-Fi LED transmitter module
+- USB cable (for programming and debug serial)
+
+### **Receiver (Pi 4)**
+- [Raspberry Pi 4 Model B (4 GB)](https://www.sparkfun.com/raspberry-pi-4-model-b-4-gb.html?src=raspberrypi)
+- Li-Fi receiver module
+
+### **Connection (UART1)**
+- **Pico TX (GPIO 4)** → **Pi 4 RX (GPIO14, Pin 8)**
+- **Pico RX (GPIO 5)** ← **Pi 4 TX (GPIO15, Pin 10)**
+
+---
+
+## Software Dependencies
+
+-   CMake ≥ 3.13
+-   ARM GCC Toolchain
+-   [Pico SDK](https://github.com/raspberrypi/pico-sdk)
+-   (Optional for Host) [iotauth](https://github.com/iotauth/iotauth) project for advanced key provisioning.
+
+
+ADD ARM sudo apt update
+```bash
+sudo apt install -y gcc-arm-none-eabi libnewlib-arm-none-eabi cmake git build-essential ninja-build
+```
+
+confirm with arm-none-eabi-gcc --version
+- if that prints go ahead and tell the pico SDK where to look now:
+```bash
+echo 'export PICO_TOOLCHAIN_PATH=/usr' >> ~/.bashrc
+source ~/.bashrc
+```
+
+
+---
+
+## Setup & Build
+
+### 1. Clone the Repository
+```bash
+git clone --recurse-submodules <repo-url>
+cd embedded
+```
+
+### 2. Configure Pico SDK
+```bash
+export PICO_SDK_PATH=/path/to/pico-sdk
+```
+
+### 3. Build the Firmware
+```bash
+mkdir build && cd build
+cmake ..
+make
+```
+
+### 4. Flash the Pico
+Connect the Pico to your computer while holding the `BOOTSEL` button. Then, copy the firmware to the Pico's USB mass storage device.
+```bash
+cp sender/lifi_flash.uf2 /media/user/RPI-RP2
+```
+
+---
+
+## How to Run
+**1. First-Time Key Provisioning:**
+   - Flash a cleared device. On the Pico's USB serial monitor, you will see the message: `No valid session key found. Waiting for one...`
+   - From a host computer connected to the Li-Fi UART, send a 32-byte secret key.
+   - The Pico will respond with `Received session key: <key>` and save it to a flash slot. The secure channel is now established.
```

Comment:
We can't say `The secure channel is now established.` just by sharing a session key. I know this is just an experimental code, but just say the two shared a session key and they can now do encrypted communication.